packages/core/utils/src/common/__tests__/transform-modules.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { MODULE_PACKAGE_NAMES, Modules } from "../../modules-sdk"
+import { transformModules } from "../define-config"
+
+describe("transformModules", () => {
+  test("convert array of modules to an object", () => {
+    const modules = transformModules([
+      {
+        resolve: require.resolve("../__fixtures__/define-config/github"),
+        options: {
+          apiKey: "test",
+        },
+      },
+    ])
+
+    expect(modules).toEqual({
+      GithubModuleService: {
+        options: {
+          apiKey: "test",
+        },
+        resolve: require.resolve("../__fixtures__/define-config/github"),
+      },
+    })
+  })
+
+  test("transform default module", () => {
+    const modules = transformModules([
+      {
+        resolve: MODULE_PACKAGE_NAMES[Modules.CACHE],
+      },
+    ])
+
+    expect(modules).toEqual({
+      cache: {
+        resolve: "@medusajs/medusa/cache-inmemory",
+      },
+    })
+  })
+
+  test("remove module when its disabled at a later stage in the array", () => {
+    const modules = transformModules([
+      {
+        resolve: MODULE_PACKAGE_NAMES[Modules.CACHE],
+      },
+      {
+        resolve: MODULE_PACKAGE_NAMES[Modules.CACHE],
+        disable: true,
+      },
+    ])
+
+    expect(modules).toEqual({})
```

Comment:
Q: if a module is disabled, I thought we expected `{ cache: false }`, the idea is that since some modules configuration can have `defaultPackage`

Reply:
Fixed in a commit by you :)